### PR TITLE
Making changes in misc.py to take some value as arguements

### PR DESCRIPTION
**Pull request by yassingh**
Consider PR 
https://github.com/hubblestack/hubblestack_data/pull/87 
for illustration of these changes.

### CODE REVIEW

File: hubblestack/files/hubblestack_nova/misc.py [hunk 1]
```diff
@@ -279,12 +279,22 @@ def world_writable_file(reason=''):
     return True if result == '' else result
 
 
-def system_account_non_login(reason=''):
+def system_account_non_login(non_login_shell='/sbin/nologin', max_system_uid='500', except_for_users=''):
```
Comment (yassingh) [inline hubblestack/files/hubblestack_nova/misc.py]: The value of non_login shell is '/sbin/nologin' in other operating systems. But in ubuntu16, the path of non login shell is '/usr/sbin/nologin'.
So, I now we are taking the path of non_login_shell as an argument. It's default value is still '/sbin/nologin', so it won't affect existing profiles.

File: hubblestack/files/hubblestack_nova/misc.py [hunk 2]
```diff
@@ -567,22 +577,30 @@ def check_all_users_home_directory(max_system_uid):
     return True if error == [] else str(error)
 
 
-def check_users_home_directory_permissions(reason=''):
+def check_users_home_directory_permissions( non_login_shell='/sbin/nologin', max_allowed_permission='750', except_for_users='' ):
```
Comment (yassingh) [inline hubblestack/files/hubblestack_nova/misc.py]: The value of non_login shell is '/sbin/nologin' in other operating systems. But in ubuntu16, the path of non login shell is '/usr/sbin/nologin'.
So, I now we are taking the path of non_login_shell as an argument. It's default value is still '/sbin/nologin', so it won't affect existing profiles.

File: hubblestack/files/hubblestack_nova/misc.py [hunk 3]
```diff
@@ -567,22 +577,30 @@ def check_all_users_home_directory(max_system_uid):
...
+    for line in cmd['stdout'].split('\n'):
+        tokens = line.split(':')
+        if tokens[0] not in users_list and tokens[6] != non_login_shell:
+            users_dirs.append(tokens[0] + " " + tokens[5])
     error = []
     for user_dir in users_dirs:
         user_dir = user_dir.split(" ")
         if len(user_dir) < 2:
-                user_dir = user_dir + [''] * (2 - len(user_dir))
+                user_dir = user_dir + ['']*(2-len(user_dir))
```
Comment (basepi) [inline hubblestack/files/hubblestack_nova/misc.py]: Add the spaces back in, please?

File: hubblestack/files/hubblestack_nova/misc.py [hunk 4]
```diff
@@ -567,22 +577,30 @@ def check_all_users_home_directory(max_system_uid):
...
         if _is_valid_home_directory(user_dir[1]):
-            result = restrict_permissions(user_dir[1], "750")
+            result = restrict_permissions(user_dir[1], max_allowed_permission)
             if result is not True:
-                error += ["permission on home directory " + user_dir[1] + " of user " + user_dir[0] + " is wrong: " + result]
+                error += ["permission on home directory " + user_dir[1]  + " of user " + user_dir[0] + " is wrong: " + result]
```
Comment (basepi) [inline hubblestack/files/hubblestack_nova/misc.py]: Unnecessary space was added.

File: hubblestack/files/hubblestack_nova/misc.py (hunk 3 above)
Comment (yassingh) [reply hubblestack/files/hubblestack_nova/misc.py]: done

File: hubblestack/files/hubblestack_nova/misc.py (hunk 4 above)
Comment (yassingh) [reply hubblestack/files/hubblestack_nova/misc.py]: removed

**Pull request merged**